cms/admin/placeholderadmin.py

```diff
@@ -579,24 +594,43 @@ def move_plugin(self, request):
                         return HttpResponseBadRequest(force_text(
                             _('parent must be in the same language as '
                               'plugin_language')))
-                    plugin.parent_id = parent.pk
-                    plugin.language = language
-                    plugin.save()
+                    plugin = plugin.update(refresh=True, parent=parent, **plugin_data)
                     plugin = plugin.move(parent, pos='last-child')
+                else:
+                    plugin = plugin.update(refresh=True, **plugin_data)
             else:
-                sibling = CMSPlugin.get_last_root_node()
-                plugin.parent = plugin.parent_id = None
-                plugin.placeholder = placeholder
-                plugin.save()
-                plugin = plugin.move(sibling, pos='right')
+                target = CMSPlugin.get_last_root_node()
+                plugin = plugin.update(refresh=True, parent=None, **plugin_data)
+                plugin = plugin.move(target, pos='right')
+
+            # Update all children to match the parent's
+            # language and placeholder
+            plugin.get_descendants().update(**plugin_data)
+
+        if order:
+            # order should be a list of plugin primary keys
+            # it's important that the plugins being referenced
+            # are all part of the same tree.
+            order = [int(pk) for pk in order]
+            plugins_in_tree = CMSPlugin.objects.filter(
+                parent=parent_id,
+                placeholder=placeholder,
+                language=language,
+                pk__in=order,
+            )
```

Comment:
Isn't this the same as ``parent.get_descendants().filter(pk__in=order)``? This might be quicker